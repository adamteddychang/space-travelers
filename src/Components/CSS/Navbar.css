@import url('https://fonts.googleapis.com/css2?family=Merriweather:wght@300;400;700&display=swap');

* {
  font-family: 'Merriweather', serif;
}


.nav-bar {
  display: flex;
  justify-content: space-between;
  margin:1rem 2rem;
}

.left-nav {
  display: flex;
  align-items: center;
}

.logo {
  width: 10%;
}

.right-nav {
  display: flex;
  align-items: center;
}

.right-nav > * {
  margin-right: 30px;
  text-decoration: none;
}

.actived-navlink {
  text-decoration: underline;
}