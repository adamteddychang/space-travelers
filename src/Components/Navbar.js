import { Link } from 'react-router-dom';
import planet from '../planet.png';

function Navbar() {
  return (
    <div className="nav-bar">
      <div className="left-nav">
        <img className="logo" src={planet} alt="logo" />
        <h1 className="header-name">Space Travelers&apos Hub</h1>
      </div>
      <div className="right-nav">
        <Link to="/">Rockets</Link>
        <Link to="/missions">Missions</Link>
        <Link to="/myprofile">My Profile</Link>
      </div>
    </div>
  );
}

export default Navbar;