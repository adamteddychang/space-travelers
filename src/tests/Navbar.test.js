import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Navbar from '../Components/Navbar';

describe('Navbar Component Test', () => {
  it('Renders NavBar', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
