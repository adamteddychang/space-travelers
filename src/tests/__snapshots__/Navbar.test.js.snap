// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Test Renders NavBar 1`] = `
<nav
  className="nav-bar"
>
  <div
    className="left-nav"
  >
    <img
      alt="logo"
      className="logo"
      src="planet.png"
    />
    <h1
      className="header-name"
    >
      Space Travelers&apos Hub
    </h1>
  </div>
  <div
    className="right-nav"
  >
    <a
      aria-current="page"
      className="nav-links actived-navlink"
      href="#/"
      onClick={[Function]}
      style={Object {}}
    >
      Rockets
    </a>
    <a
      aria-current={null}
      className="nav-links"
      href="#/missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      aria-current={null}
      className="nav-links"
      href="#/myprofile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </div>
</nav>
`;
